ledger-core/virtual/integration/memory_request_test.go

```diff
@@ -0,0 +1,263 @@
+	"github.com/insolar/assured-ledger/ledger-core/testutils/synchronization"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/descriptor"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/execute"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/integration/utils"
+)
+
+func TestVirtual_VCachedMemoryRequest(t *testing.T) {
+	insrail.LogCase(t, "")
+	defer commonTestUtils.LeakTester(t)
+	testCases := []struct {
+		name        string
+		method      bool
+		constructor bool
+		pending     bool
+	}{
+		{name: "Object state created from constructor", constructor: true},
+		{name: "Object state created from method", method: true},
+		{name: "Object state created from pending", pending: true},
+	}
+
+	const (
+		newState     = "new state"
+		methodState  = "method state"
+		pendingState = "pending state"
+	)
+
+	for _, cases := range testCases {
+		t.Run(cases.name, func(t *testing.T) {
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			runnerMock := logicless.NewServiceMock(ctx, t, nil)
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+			var (
+				classA        = server.RandomGlobalWithPulse()
+				objectGlobal  = reference.NewSelf(classA.GetLocal())
+				prevPulse     = server.GetPulse().PulseNumber
+				expectedState string
+			)
+
+			server.IncrementPulse(ctx)
+
+			if cases.constructor {
+
+				pl := utils.GenerateVCallRequestConstructor(server)
+				pl.Caller = classA
+				callOutgoing := pl.CallOutgoing
+
+				result := requestresult.New([]byte(newState), objectGlobal)
+
+				key := callOutgoing.String()
+				runnerMock.AddExecutionMock(key).
+					AddStart(nil, &execution.Update{
+						Type:   execution.Done,
+						Result: result,
+					})
+				runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+					Interference: pl.CallFlags.GetInterference(),
+					State:        pl.CallFlags.GetState(),
+				}, nil)
+
+				server.SendPayload(ctx, pl)
+				expectedState = newState
+
+			} else {
+				Method_PrepareObject(ctx, server, payload.StateStatusReady, objectGlobal, prevPulse)
+			}
+
+			// call method
+			if cases.method {
+				pl := utils.GenerateVCallRequestMethod(server)
+				pl.Callee = objectGlobal
+				pl.CallSiteMethod = "ordered"
+				callOutgoing := pl.CallOutgoing
+
+				result := requestresult.New([]byte(methodState), objectGlobal)
+
+				key := callOutgoing.String()
+				runnerMock.AddExecutionMock(key).
+					AddStart(nil, &execution.Update{
+						Type:   execution.Done,
+						Result: result,
+					})
+				runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+					Interference: pl.CallFlags.GetInterference(),
+					State:        pl.CallFlags.GetState(),
+				}, nil)
+
+				server.SendPayload(ctx, pl)
+				expectedState = methodState
+			}
+
+			// pending
+			if cases.pending {
```

Comment:
If for pending test is way too big, I prefer to have separate test for it.

Reply:
will be done later